File: nicevideoplayerdemo/src/main/java/com/example/nicevideoplayerdemo/BaseActivity.java
package com.example.nicevideoplayerdemo;

import androidx.appcompat.app.AppCompatActivity;

import com.xiao.nicevideoplayer.NiceVideoPlayerManager;

/**
 * User : Blues
 * Date : 2019/3/5
 * Time : 14:59
 * Email : huajianlan@rastar.com
 */

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onStop() {
        super.onStop();
        NiceVideoPlayerManager.instance().releaseNiceVideoPlayer();
    }

    @Override
    public void onBackPressed() {
        if (NiceVideoPlayerManager.instance().onBackPressd()) return;
        super.onBackPressed();
    }
}

File: kotlindemo/src/main/java/com/example/kotlindemo/BaseActivity.kt
package com.example.kotlindemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

/**
 * User : Blues
 * Date : 2019/10/11
 * Time : 15:46
 */

abstract class BaseActivity : AppCompatActivity() {

    private var mContext: BaseActivity? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mContext = this

        setContentView(setLayoutId())

        setListener()
    }


    abstract fun setLayoutId(): Int

    abstract fun setListener()

}